.github/workflows/integration-test.yml

```
@@ -0,0 +1,53 @@
+name: Integration test
+
+on:
+  schedule:
+    - cron: "0 0 * * *"
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+# Cancel the job if new commits are pushed
+# https://stackoverflow.com/questions/66335225/how-to-cancel-previous-runs-in-the-pr-when-you-push-new-commitsupdate-the-curre
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  test:  
+    strategy:
+      fail-fast: false
+      matrix:
+        python: ["3.10", "3.12"]
+        package: ["mudata", "spatialdata", "scirpy", "muon", "scanpy", "squidpy", "scvi-tools"]
+
+    defaults:
+      run:
+        shell: bash -el {0}
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          repository: scverse/${{ matrix.package }}
+  
+      - name: Install Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python }}
+
+      - name: Install UV
+        uses: hynek/setup-cached-uv@v2
+
+      - name: Install AnnData + test dep
+        run: |
+          printf "numba>=0.60.0\nnumpy<2.0" | tee /tmp/constraints.txt
+          uv pip install --compile --system ".[test]" pytest-xdist -c /tmp/constraints.txt -v 
+          uv pip install --compile --system git+https://github.com/scverse/anndata 
```

please don’t do that! only one `pip` or `uv pip` call per environment or nothing is guaranteed!

---

Ah, so you intentionally want to disregard the dependency specification here.

What keeps `uv` from deciding the following?

> well, you asked to install a specific version of `anndata`, so I’m removing everything that is incompatible with that version to honor your request

If there is no documentation about how this kind of conflict is dealt with, I feel like we should either deliberately patch out the pins or just set a constraint for the newest anndata and fail the test at the resolution stage instead.

---

> how would uv honor scirpy requiring anndata<0.10.9 while also installing our main branch?

it can’t, so there are two possibilities:
1. uv documents how it acts in case of conflicts like that, and we’re happy with that algorithm.
2. uv just does whatever, which might include uninstalling the scirpy version from git and installing e.g. some old scirpy version that doesn‘t have a pin.

I feel like the latter is probably the case, so we should do this

```suggestion
          uv pip install --compile --system -v ".[test]" pytest-xdist -c /tmp/constraints.txt "anndata @ git+https://github.com/scverse/anndata"
```

---

I think for now, once `scirpy` upgrades I will move this into the same line.  I think patching is bad,because going forward installation will error if someone tries to upper pin anndata, and we would want to know that because they shouldn't